### PR TITLE
Filter join conditions #2198

### PR DESCRIPTION
**Pull request by zahardev**
Adds a filter for join conditions. Most immediately, this is required for https://github.com/GravityKit/Multiple-Forms/pull/105.

💾 [Build file](https://www.dropbox.com/scl/fi/rgwjagesqw34naf4rkgw6/gravityview-2.31.1-753893103.zip?rlkey=qphdqt1tkm7nd0pmiev3xx8bs&dl=1) (753893103).

### CODE REVIEW

File: future/includes/class-gv-view.php [hunk 1]
```diff
@@ -1250,6 +1250,8 @@ public function get_entries( $request = null ) {
 						)
 					);
 
+					$status_conditions = apply_filters( 'gk/gravityview/view/entries/join_conditions', $status_conditions, $join, $this );
```
Comment (zackkatz) [inline future/includes/class-gv-view.php:1253]: @zahardev Please add full docblocks for filters. See line 1137 and 1556 for an example.

**Review by zackkatz — CHANGES_REQUESTED** [review]

File: future/includes/class-gv-view.php (hunk 1 above)
Comment (zahardev) [reply future/includes/class-gv-view.php:1253]: @zackkatz Could you please suggest the filter name here? Which is better - `gk/gravityview/view/entries-join-conditions`, gk/gravityview/view/entries/join-conditions` or something else?

**Comment by mrcasual** [comment]
@zahardev, thanks for the fixes. Please fix the merge conflict and then hit that "merge" button.

**Comment by rafaehlers** [comment]
FIY this has been tested: https://github.com/GravityKit/Multiple-Forms/pull/105#issuecomment-2477383089

**Comment by zahardev** [comment]
@mrcasual 

Yes, we need that part for the strict Strict Entry Match view.

**Pull request merged**